dsgrid/dimension/time_utils.py

```diff
@@ -0,0 +1,505 @@
+    if timezone is None:
+        timezone = dim_model.get_tzinfo()
+    if data_adjustment is None:
+        data_adjustment = DataAdjustmentModel()
+    dt_ranges = time_dimension_config._create_represented_time_ranges()
+    ranges = []
+    time_ranges = build_time_ranges(
+        dt_ranges, dim_model.str_format, model_years=model_years, tz=timezone
+    )
+    for index_range, time_range in zip(dim_model.ranges, time_ranges):
+        ranges.append(
+            make_time_range(
+                start=time_range[0],
+                end=time_range[1],
+                frequency=dim_model.frequency,
+                data_adjustment=data_adjustment,
+                time_interval_type=dim_model.time_interval_type,
+                start_index=index_range.start,
+            )
+        )
+
+    return ranges
+
+
+def list_timestamps(
+    time_dimension_config,  #: DateTimeDimensionConfig,
+    model_years: Optional[list[int]] = None,
+    timezone: TimeZone = None,
+    data_adjustment: DataAdjustmentModel = None,
+):
+    timestamps = []
+    for time_range in get_time_ranges(
+        time_dimension_config,
+        model_years=model_years,
+        timezone=timezone,
+        data_adjustment=data_adjustment,
+    ):
+        timestamps += [DatetimeTimestampType(x) for x in time_range.list_time_range()]
+    return timestamps
+
+
+def list_time_indices(
+    time_dimension_config,  #: IndexTimeDimensionConfig,
+    model_years: Optional[list[int]] = None,
+    timezone: TimeZone = None,
+    data_adjustment: DataAdjustmentModel = None,
+):
+    indices = []
+    for index_range in get_index_ranges(
+        time_dimension_config,
+        model_years=model_years,
+        timezone=timezone,
+        data_adjustment=data_adjustment,
+    ):
+        indices += [IndexTimestampType(x) for x in index_range.list_time_range()]
+    return indices
+
+
+def build_index_time_map(
+    time_dimension_config,  #: IndexTimeDimensionConfig,
+    model_years=None,
+    timezone=None,
+    data_adjustment=None,
+):
+
+    time_col = time_dimension_config.get_load_data_time_columns()
+    assert len(time_col) == 1, time_col
+    time_col = time_col[0]
+    indices = list_time_indices(
+        time_dimension_config,
+        model_years=model_years,
+        timezone=timezone,
+        data_adjustment=data_adjustment,
+    )
+    timestamps = list_timestamps(
+        time_dimension_config,
+        model_years=model_years,
+        timezone=timezone,
+        data_adjustment=data_adjustment,
+    )
+    ts_time_col = timestamps[0]._fields[0]
+    schema = StructType(
+        [
+            StructField(time_col, IntegerType(), False),
+            StructField(ts_time_col, TimestampType(), False),
+        ]
+    )
+    data = []
+    for a, b in zip(indices, timestamps):
+        data.append((a[0], b[0]))
+    df_time = get_spark_session().createDataFrame(data, schema=schema)
+
+    return df_time
+
+
+def build_datetime_dataframe(
```

Comment:
Not used. Do we need it?

Reply:
Can we wait until Industrial time has been submitted to DECARB?